src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -59,6 +62,12 @@
 
   private static final int MAXIMUM_ALLOWED_UPDATE_ACTIONS = 500;
   static final String CREATE_FAILED = "Failed to create draft with key: '%s'. Reason: %s";
+  /*
+   * A key is a 41 characters long hashed string. (i.e: c25a67e262265fbc36edb33ed0375263586be278) We
+   * chunk them in 250 keys we will have around a query around 11.000 characters. Above this size it
+   * could return - Error 413 (Request Entity Too Large)
+   */
+  static final int CHUNK_SIZE = 250;
```

Comment:
Maybe we should be more conservative here as AFAIK the request length is calculated not just from the 250 keys but also all of the provided headers, encoded parameters and request URL.

Reply:
Yes @butenkor , But as mentioned in the PR comment already, I wasn't sure how to calculate the key size here. (I see this methods in BaseService uses Key and Sku values. When I was creating ProductType, It says the below message - 
 "May only contain between 2 and 256 alphanumeric characters, underscores, or hyphens (no white space or special characters like ñ, ü, #, %) ". 
 But I don't see any restrictions for "key" values while creating Products and Categories.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/688/commits/6c5a1d96da9ff116e189543315e418cfcca67813.